### PR TITLE
Update Nucleo-64 boards to fit configuration guidelines

### LINKED ISSUE #9246: Get ST Nucleo boards compliant with default configuration guidelines
Created from #7151, dedicated to ST Nucleo boards

Following introduction of "default board configuration guidelines" (#6858), one part of the work is to move boards already ported to zephyr to conform with guidelines available here:
http://docs.zephyrproject.org/porting/board_porting.html#default-board-configuration

https://github.com/zephyrproject-rtos/zephyr/tree/master/boards/arm/nucleo_f429zi could be used as reference for this update.

### PR DESCRIPTION
**Pull request by erwango**
Fixes #9246 for Nucleo-64 boards

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
doc changes LGTM

File: boards/arm/nucleo_f091rc/doc/nucleof091rc.rst [hunk 1]
```diff
@@ -121,20 +121,14 @@ Board connectors:
...
-- SPI1_SCK : PB3
-- SPI1_MISO : PB4
-- SPI1_MOSI : PB5
-- SPI2_SCK : PB13
-- SPI2_MISO : PB14
-- SPI2_MOSI : PB15
+- UART_1 TX/RX : PB6/PB7
+- UART_2 TX/RX : PA2/PA3 (ST-Link Virtual COM Port)
+- I2C1 SCL/SDA : PB8/PB9 (Arduino I2C)
+- I2C2 SCL/SDA : PA11/PB12
```
Comment (MaureenHelm) [inline boards/arm/nucleo_f091rc/doc/nucleof091rc.rst]: PA12?

Comment (erwango) [reply boards/arm/nucleo_f091rc/doc/nucleof091rc.rst]: Right, txs! Fixed

**Comment by erwango** [comment]
@MaureenHelm, can you have a new check?

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**